dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java

```diff
@@ -268,74 +269,104 @@ protected Identifier createNewIdentifier(Versionable versionable, Folder folder)
 	    return createNewIdentifier(versionable,folder,UUIDGenerator.generateUuid());
 	}
 
+	@Override
+	protected Identifier createNewIdentifier(final Folder folder, final Folder parent, final String existingId) throws DotDataException {
+		final User systemUser = APILocator.getUserAPI().getSystemUser();
+		final Identifier identifier = new Identifier();
+
+		identifier.setId(existingId!=null?existingId:UUIDGenerator.generateUuid());
```

Comment:
I guess it is generateUuid instead of deterministic, because the existingId was already deterministic right? or should we use deterministic here?

Reply:
That's correct! The generateUUID will be used as a fallback to avoid collisions with existing deterministic IDs